tests/Fixer/LanguageConstruct/UnionNullFixerTest.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\LanguageConstruct;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\LanguageConstruct\UnionNullFixer
+ *
+ * @requires PHP 8.0
+ */
+final class UnionNullFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideReturnTypeCases
+     */
+    public function testReturnType(string $expected, ?string $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public static function provideReturnTypeCases(): iterable
+    {
+        yield [
+            '<?php function foo(): string|null {} ?>',
```

Comment:
Any particular reason why not `null|string`? Wouldn't that be easier to implement?

You need to add an integration test for [`ordered_types`](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/rules/class_notation/ordered_types.rst) anyway.

Reply:
Personal preference I suppose. Not like putting it at the end is particularly hard, so...

I'll look into it, if you could point me to an existing example of multiple fixer interaction tests that'd help out a ton.